GEECS-PythonAPI/geecs_python_api/controls/data_acquisition/multiscan_analysis.py

```diff
@@ -0,0 +1,294 @@
+            template_dir = os.path.join(item['path_scan'], template)
+            path_list = list(glob.glob(template_dir))
+
+            # iterate through path list for item of interest
+            for val in os.listdir(item['path_scan']):
+                test_path = os.path.join(item['path_scan'], val)
+
+                # if path is a directory and satisfies template
+                # then record to scan dict and break loop
+                if (os.path.isdir(test_path) and test_path in path_list):
+                    scan_dict[key]['device'] = val
+                    break
+
+        return scan_dict
+
+    def scan_analysis_backup(self, scan, scan_parameter=None):
+
+        # load sfile
+        sfile = super().load_sfile(scan['path_sfile'])
+
+        # identify scan parameter bin values
+        bins = sfile[scan_parameter].unique()
+
+        # identify shots for each bin
+        binned_shots = {val: sfile.loc[sfile[scan_parameter] == val, 'Shotnumber'].tolist() for val in bins}
+
+        # for each bin, load images and average
+
+        pass
+
+    def scan_analysis(self, scan):
+
+        # initialize VisaEBeamAnalysis
+        analyzer = VisaEBeamAnalysis(scan['path_scan'], scan['device'])
+        self.scan_parameter = analyzer.scan_parameter
+
+        # get blue diode coordinates
+        analysis_settings = analyzer.camera_analysis_configs[scan['device']]
+        scan['diode_coords'] = (analysis_settings['Blue Centroid X'] - analysis_settings['Left ROI'],
+                                analysis_settings['Blue Centroid Y'] - analysis_settings['Top ROI'])
+
+        # perform visa e beam analysis
+        binned_images = analyzer.perform_bulk_image_analysis()
+
+        # record in scan dict
+        scan['binned_data'] = {ind+1: {'value': val,
+                                       'image': binned_images[ind]}
+                               for ind, val in enumerate(analyzer.binned_param_values)}
+
+        return scan
+
+    def load_analysis_results(self, scan):
+        """
+        Load existing analysis images rather than perform full analysis.
+
+        Needs to be revisited once merged.
+        """
+
+        # initialize VisaEBeamAnalysis
+        analyzer = VisaEBeamAnalysis(scan['path_scan'], scan['device'])
+        self.scan_parameter = analyzer.scan_parameter
+
+        # get bin numbers and values (assumes correlated bin number and value!!)
+        bin_num = np.unique(analyzer.bins)
+        bin_vals = analyzer.binned_param_values
+
+        # load already processed images
+        img = cv2.imread('Z:/data/Undulator/Y2024/10-Oct/24_1031/scans/Scan053/UC_VisaEBeam1_1_processed.png', cv2.IMREAD_UNCHANGED)
+        scan['binned_data'] = {val: {'value': bin_vals[ind],
+                                     'image': read_imaq_image(analyzer.scan_directory / f"{scan['device']}_{val}_processed.png")}
+                               for ind, val in enumerate(bin_num)}
+
```

Comment:
Down-the-road comment:  could be cool if there was a way to generalize this to all multiscan analysis routines and put in the parent.  I could see us rewriting this kind of function often